File: backend/src/main/java/com/campus/Campus_Connect/features/event/dto/response/EventHistoryItemResponse.java
package com.campus.Campus_Connect.features.event.dto.response;

import lombok.*;

import java.time.Instant;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EventHistoryItemResponse {

    private Integer id;

    private String title;

    private String venue;

    private Instant startTime;

    private Instant endTime;

    private String posterUrl;

}

File: android-app/app/src/main/java/com/example/campusconnect/feature/events/data/remote/response/EventHistoryItemResponse.kt
package com.example.campusconnect.feature.events.data.remote.response

import com.google.gson.annotations.SerializedName

data class EventHistoryItemResponse(
    @SerializedName("id")
    val id: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("venue")
    val venue: String,

    @SerializedName("startTime")
    val startTime: String,

    @SerializedName("endTime")
    val endTime: String?,

    @SerializedName("posterUrl")
    val posterUrl: String?
)